src/Microsoft.Android.Sdk.ILLink/TypeMappingStep.cs

```diff
@@ -15,10 +19,27 @@ namespace Microsoft.Android.Sdk.ILLink;
 public class TypeMappingStep : BaseStep
 {
 	const string AssemblyName = "Microsoft.Android.Runtime.NativeAOT";
-	const string TypeName = "Microsoft.Android.Runtime.NativeAotTypeManager";
+	const string TypeName = "Microsoft.Android.Runtime.TypeMapping";
+	const string SystemIOHashingAssemblyPathCustomData = "SystemIOHashingAssemblyPath";
 	readonly IDictionary<string, List<TypeDefinition>> TypeMappings = new Dictionary<string, List<TypeDefinition>> (StringComparer.Ordinal);
 	AssemblyDefinition? MicrosoftAndroidRuntimeNativeAot;
 
+	AssemblyLoadContext? _hashingAssemblyLoadContext;
+	delegate ulong HashMethod (ReadOnlySpan<byte> data, long seed = 0);
+	HashMethod? _hashMethod;
+
+	protected override void Process ()
+	{
+		if (!Context.TryGetCustomData (SystemIOHashingAssemblyPathCustomData, out var assemblyPath)) {
+			throw new InvalidOperationException ($"The {nameof (TypeMappingStep)} step requires setting the '{SystemIOHashingAssemblyPathCustomData}' custom data");
+		} else if (!System.IO.File.Exists (assemblyPath)) {
+			throw new InvalidOperationException ($"The '{SystemIOHashingAssemblyPathCustomData}' custom data must point to a valid assembly path ('{assemblyPath}' does not exist)");
+		}
+
+		_hashingAssemblyLoadContext = new AssemblyLoadContext (name: null, isCollectible: true);
```

Comment:
If you do not come up with something better, you can just use `Assembly.LoadFile` here. `Assembly.LoadFile` does exactly what you are doing here: load an assembly into its own isolated AssemblyLoadContext.